pulp_smash/tests/rpm/api_v2/test_sync_publish.py

```diff
@@ -438,3 +473,85 @@ def test_unit_integrity(self):
         for i, module in enumerate(self.rpms[1:]):
             with self.subTest(i=i):
                 self.assertEqual(self.rpms[0], module)
+
+
+class SyncOnDemandTestCase(_BaseTestCase):
+    """Assert Yum supports on-demand syncing."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an RPM repository with a valid feed and sync it.
+
+        Do the following:
+
+        0. Reset Pulp, including the Squid cache.
+        1. Sync/publish a repository using the 'on_demand' download policy.
+        2. Download an RPM from the published repository.
+        3. Download the same RPM to ensure it is served by the cache.
+        """
+        super(SyncOnDemandTestCase, cls).setUpClass()
+
+        # Required to ensure the `locally_stored_units` is 0 before we start.
+        utils.reset_squid(cls.cfg)
+        utils.reset_pulp(cls.cfg)
+        importer_config = {
+            'feed': RPM_FEED_URL,
+            'download_policy': 'on_demand',
+        }
+        repo, tasks = _sync_and_publish_repo(
+            cls.cfg,
+            importer=importer_config
+        )
+        client = api.Client(cls.cfg, api.echo_handler)
+        cls.repo = client.get(repo['_href'], params={'details': True}).json()
+
+        url = urljoin(
+            '/pulp/repos/',
+            repo['id'] + '/',
+        )
+        url = urljoin(url, RPM)
+        cls.rpm = client.get(url)
+        cls.same_rpm = client.get(url)
+
+    def test_local_units(self):
+        """Assert no content units were downloaded besides metadata."""
+        metadata_units = filter(
+            lambda unit_tuple: unit_tuple[0] not in ('rpm', 'drpm', 'srpm'),
+            self.repo['content_unit_counts'].items()
+        )
+        metadata_unit_count = sum([unit[1] for unit in metadata_units])
+        self.assertEqual(self.repo['locally_stored_units'], metadata_unit_count)
+
+    def test_repository_units(self):
+        """Assert there is at least one content unit in the repository."""
+        total_units = sum(self.repo['content_unit_counts'].values())
+        self.assertEqual(self.repo['total_repository_units'], total_units)
+
+    def test_request_history(self):
+        """Assert that the initial request received a 302 Redirect."""
+        self.assertTrue(self.rpm.history[0].is_redirect)
+
+    def test_rpm_checksum(self):
+        """Assert the checksum of the downloaded RPM matches the metadata."""
+        checksum = hashlib.sha256(self.rpm.content).hexdigest()
+        self.assertEqual(RPM_SHA256_CHECKSUM, checksum)
+
+    def test_rpm_cache_lookup_header(self):
+        """Assert the first request resulted in a cache miss from Squid."""
+        self.assertIn('MISS', self.rpm.headers['X-Cache-Lookup'])
+
+    def test_rpm_cache_control_header(self):
+        """Assert that the request has the Cache-Control header set."""
+        self.assertEqual(
+            'public, s-maxage=86400, max-age=86400',
```

Comment:
Can these headers legally appear in some other order? That is, can the server legally return one of the following?

* `public, max-age=86400, s-maxage=86400`
* `s-maxage=86400, max-age=86400, public`
* …another seven combinations